reactor-netty-http/src/main/java/reactor/netty/http/client/WebsocketClientOperations.java

```diff
@@ -223,8 +223,7 @@ public NettyOutbound send(Publisher<? extends ByteBuf> dataStream) {
 			return FutureMono.deferFuture(() -> {
 				if (CLOSE_SENT.getAndSet(this, 1) == 0) {
 					discard();
-					onCloseState.emitValue(new WebSocketCloseStatus(frame.statusCode(), frame.reasonText()),
-							Sinks.EmitFailureHandler.FAIL_FAST);
+					onCloseState.tryEmitValue(new WebSocketCloseStatus(frame.statusCode(), frame.reasonText()));
```

Comment:
I don't think that was what @rstoyanchev was advocating for. unlike with `tryEmitEmpty`, it's riskier to ignore the `EmitResult` of `tryEmitValue`. Have you gone through the possible failure modes and made sure all of them can either be ignored or are guaranteed not to occur?

I suspect the `CLOSE_SENT` guard protects against most (all) possible failures, but in that case be sure to again put a comment that `EmitResult` is ignored on purpose. maybe even go as far as detailing the ignored failure modes (so that if one is added later, it forces us to reason about this piece of code again)

Reply:
This looks good to me. It's already synchronized so it can't be non-serialized, it can't overflow, and it doesn't matter if it's cancelled.

Reply:
`FAIL_TERMINATED` - `CLOSE_SENT` guarantees only one emission will happen
`FAIL_OVERFLOW`, `FAIL_NON_SERIALIZED`, `FAIL_ZERO_SUBSCRIBER` - should not be a problem here?
`FAIL_CANCELLED` -  what to do here?

Reply:
Whoever is listening for CloseStatus, cancelled, which means they no longer care, so I'd say nothing to do. FWIW we have a [similar case](https://github.com/spring-projects/spring-framework/blob/e1f51b4bf80c7284f7c6ad918385a24b680b6549/spring-webflux/src/main/java/org/springframework/web/reactive/socket/adapter/AbstractListenerWebSocketSession.java#L219) in the Spring Framework.

Reply:
done

Reply:
great 👍